python/pecos/qeclib/steane/steane_class.py

```diff
@@ -97,117 +97,46 @@ def __init__(self, name: str, default_rus_limit: int = 3, ancillas: QReg | None
 
         self.default_rus_limit = default_rus_limit
 
-    def p(self, state: str, rus_limit: int | None = None):
+    def p(self, state: str, reject: Bit | None = None, rus_limit: int | None = None):
         """Prepare a logical qubit in a logical Pauli basis state."""
-        if rus_limit is None:
-            rus_limit = self.default_rus_limit
-
-        match state:
-            case "+X" | "X":
-                return self.px(rus_limit=rus_limit)
-            case "-X":
-                return self.pnx(rus_limit=rus_limit)
-            case "+Y" | "Y":
-                return self.py(rus_limit=rus_limit)
-            case "-Y":
-                return self.pny(rus_limit=rus_limit)
-            case "+Z" | "Z":
-                return self.pz(rus_limit=rus_limit)
-            case "-Z":
-                return self.pnz(rus_limit=rus_limit)
-            case _:
-                msg = f"State {state} is not implemented!"
-                raise NotImplementedError(msg)
-
-    def px(self, rus_limit: int | None = None):
-        """Prepare logical |+X>, a.k.a. |+>"""
-
-        if rus_limit is None:
-            rus_limit = self.default_rus_limit
+        if state in ["X", "Y", "Z"]:
+            state = f"+{state}"
```

Comment:
The match/case here was redundant with that in in PrepRUS -> LogZeroRot, [here](https://github.com/PECOS-packages/PECOS/blob/development/python/pecos/qeclib/steane/preps/pauli_states.py#L153-L179).

Reply:
As minor cleanup, I also made functions like `Steane.pz/pnz/px/etc.` call `Steane.p`, and likewise with `Steane.mx/my/mz` and `Steane.m`.

Reply:
Incidentally, @ciaranra would you want to make "X", "Y", and "Z" recognized as "+X", "+Y", and "+Z" in [`LogZeroRot`](https://github.com/PECOS-packages/PECOS/blob/a0989e7aec73dadbbf610a6b75ceb1df66db8c4b/python/pecos/qeclib/steane/preps/pauli_states.py#L153-L179) as well?

Reply:
Yeah, that would be good to do to keep things consistent.

Reply:
done 🙂